src/core/internal/dassert.d

```diff
@@ -88,7 +88,15 @@ private string miniFormat(V)(const ref V v)
     else static if (__traits(isFloating, V))
     {
         char[60] val;
-        const len = sprintf(&val[0], "%g", v);
+        int len;
+        static if (is(V == cfloat) || is(V == cdouble))
+            len = sprintf(&val[0], "%gi", v.re, v.im);
```

Comment:
I literally meant `"%g + %gi"` - you are simply printing 2 separately passed doubles => `1.5 + 0.5i`.